### ISSUE TITLE
Optional params to Parameterized View?

### ISSUE THREAD

**Issue by goenning** [ClickHouse/ClickHouse#53004]
I want to build a view that has an optional date filter, but I just can't seem to find a way to do it.

My only attempt was adding a `OR {param} IS NULL` condition which I think it would work, but it seems like I can't use the same param twice.

The view below gives me this error:

```
Code: 15. DB::Exception: Column 'p_date_from' already exists: While processing (app_id = {p_app_id:String}) AND (timestamp >= ({p_date_from:String} OR ({p_date_from:String} IS NULL))). (DUPLICATE_COLUMN) (version 23.3.8.21 (official build))
```

Example:
```
CREATE OR REPLACE VIEW get_data
AS

    SELECT *
    FROM events
    PREWHERE app_id = {p_app_id:String}
    AND (timestamp >= {p_date_from:String} OR {p_date_from:String} IS NULL)
```

Any other suggestion on how to do this?

**Comment by melvynator** [comment]
Seems like this feature request: https://github.com/ClickHouse/ClickHouse/issues/49661 

Can you confirm?

**Comment by goenning** [comment]
No, it's a different thing. That request is for a dynamic column, this one seems more like a bug when I look at it. It should be possible to use a variable twice within the view

**Comment by pkit** [comment]
@goenning I think the bug was fixed I still see it in CH 23.4.x but it works in CH >=23.5.1

**Comment by nikitamikhaylov** [comment]
Also it works perfectly in `24.4` https://fiddle.clickhouse.com/00edb753-f307-4ca4-8ff8-7864ce97137d
